crates/ruff_cli/src/diagnostics.rs

```diff
@@ -177,7 +177,18 @@ impl AddAssign for FixMap {
     }
 }
 
+fn get_override_source_type(
+    path: Option<&Path>,
+    extension: &ExtensionMapping,
+) -> Option<PySourceType> {
+    let Some(ext) = path.and_then(|p| p.extension()).and_then(|p| p.to_str()) else {
+        return None;
+    };
+    extension.map_ext(ext).map(PySourceType::from)
+}
```

Comment:
I believe this can be reduced to:

```rust
fn override_source_type(path: Option<&Path>, extension: &ExtensionMapping) -> Option<PySourceType> {
    let ext = path?.extension()?.to_str()?;
    extension.map_ext(ext).map(PySourceType::from)
}
```